src/include/migraphx/op/where.hpp

```diff
@@ -42,9 +42,19 @@ struct where
 
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(3).same_dims();
+        check_shapes{inputs, *this, true}.has(3).same_dims();
         auto s1 = inputs.at(1);
         auto s2 = inputs.at(2);
+        if(s1.dynamic() or s2.dynamic())
```

Comment:
This would mean it handles one static and one dynamic. we don't know which one is which and it always return `s1`.

Reply:
ref. my answers below.  If either shape is dynamic, then the other has to be exactly equal (including being dynamic)